improver_tests/acceptance/test_checksums.py

```diff
@@ -30,17 +30,66 @@
 # POSSIBILITY OF SUCH DAMAGE.
 """Bulk checking and updating of checksum file"""
 
+import difflib
+import functools
+import locale
 import os
 import pathlib
+from contextlib import contextmanager
+from typing import Callable, Dict, List
 
 import pytest
 
 from . import acceptance as acc
 
-pytestmark = [pytest.mark.acc, acc.skip_if_kgo_missing]
 
+@contextmanager
+def temporary_sort_locale(collate: str) -> Callable[[str, str], bool]:
+    """
+    Set a temporary locale for sorting.
 
+    Args:
+        collate: LC_COLLATE locale name
+
+    Yields:
+        String comparison function
+    """
+    old = locale.getlocale(locale.LC_COLLATE)
+    try:
+        locale.setlocale(locale.LC_COLLATE, collate)
+        yield locale.strcoll
+    finally:
+        locale.setlocale(locale.LC_COLLATE, old)
+
+
+def checksums_to_text(
+    path_csums: Dict[pathlib.Path, str], sort: bool = False
+) -> List[str]:
+    """
+    Convert checksum dict to sha256sum like format.
+
+    Args:
+        path_csums: per-file checksums
+        sort: apply sorting to file paths
+
+    Returns:
+        List of checksum file lines, no newline at end.
+    """
+    path_csums = {str(path): csum for path, csum in path_csums.items()}
+    if sort:
+        paths = [str(path) for path in path_csums.keys()]
```

Comment:
I'm not sure that you need the `str` as the keys of `path_csums` are converted to strings above.

Reply:
In which case the if block can be rewritten:

```python
    if sort:
        # sorting uses C locale to avoid locale-specific variation
        with temporary_sort_locale("C") as strcoll:
            paths = sorted(path_csums.keys(), key=functools.cmp_to_key(strcoll))
```

Reply:
Applied suggested change